apps/etl/transform/sources/glide.py

```diff
@@ -16,35 +27,32 @@ def transform_glide_event_data(data):
     data_file_path = glide_instance.resp_data.path  # Absolute file path
     try:
         with open(data_file_path, "r") as file:
-            data = file.read()
+            data = json.loads(file.read())
     except FileNotFoundError:
         logger.error(f"File not found: {data_file_path}")
         raise
     except IOError as e:
         logger.error(f"I/O error while reading file: {str(e)}")
         raise
 
-    transformer = GlideTransformer(GlideDataSource(source_url=glide_instance.url, data=data))
-
-    try:
-        with transaction.atomic():
-            transformed_event_item = transformer.make_items()
-            if not transformed_event_item == []:
-                for item in transformed_event_item:
-
-                    GlideTransformation.objects.create(
-                        extraction=glide_instance,
-                        data=item.to_dict(),
-                        status=GdacsTransformation.TransformationStatus.SUCCESS,
-                        failed_reason="",
-                    )
-                    logger.info("Transformation ended for glide event data")
-                return {"data": [i.to_dict() for i in transformed_event_item]}
-    except Exception as e:
-        logger.info(f"Glide transformation failed for extraction id {glide_instance.id}")
-        GlideTransformation.objects.create(
-            extraction=glide_instance,
-            status=GdacsTransformation.TransformationStatus.FAILED,
-            failed_reason=str(e),
-        )
-        raise
+    if data["glideset"]:
+        try:
+            transformer = GlideTransformer(GlideDataSource(source_url=glide_instance.url, data=json.dumps(data)))
+            with transaction.atomic():
+                transformed_event_item = transformer.make_items()
+                if not transformed_event_item == []:
+                    for item in transformed_event_item:
+                        item_type = glide_item_type_map[item.collection_id]
+
+                        GlideTransformation.objects.create(
+                            extraction=glide_instance,
+                            data={"item": item.to_dict(), "collection_id": item.collection_id},
+                            item_type=item_type,
+                            status=TransformationStatus.SUCCESS,
+                            failed_reason="",
+                        )
+                        logger.info("Transformation ended for glide event data")
+                    return {"data": [i.to_dict() for i in transformed_event_item]}
+        except Exception as e:
+            logger.info(f"Glide transformation failed for extraction id {glide_instance.id}, {e}")
```

Comment:
Handle this error

Update the transformation status to failed
logger.error without f, use extra for the id and use exc_info=True
We can use SOURCE with f